### PR TITLE
Up coming alert feature

### PR DESCRIPTION
**Pull request by Huggenguggen**
Add Alert functionality.

On startup, app will check if there are any alerts.

Change Tasks such that they store the alert window as an attribute.

User can change alert timing with a command.

### CODE REVIEW

File: src/main/java/seedu/task/model/task/Event.java [hunk 1]
```diff
@@ -57,6 +58,16 @@ public boolean equals(Object other) {
...
     }
 
+    /**
+     * Returns true if {@code to} is within the {@code alertWindow} from the current time.
+     */
+    @Override
+    public boolean isComingUp() {
+        boolean isAfterNow = this.to.value.isAfter(LocalDateTime.now());
+        boolean isBeforeAlert = this.to.value.minus(alertWindow).isBefore(LocalDateTime.now());
+        return isAfterNow && isBeforeAlert;
```
Comment (lywich) [inline src/main/java/seedu/task/model/task/Event.java:68]: I would like to suggest checking for an event "from" time as well. An example use case would be an event from 6pm to 8pm. A user checks the app at 5pm with "alert 2" and will be able to see that there is a dinner starting soon.

**Review by lywich — COMMENTED** [review]
Great work as always. Left a comment for your consideration to make the feature more complete.

File: src/main/java/seedu/task/model/task/Event.java (hunk 1 above)
Comment (Huggenguggen) [reply src/main/java/seedu/task/model/task/Event.java:68]: Great suggestion! I will add that check.

**Review by lywich — APPROVED** [review]
LGTM, great work.

**Pull request merged**